assets/app/features/upgrade/controllers/upgrade-nodes.controller.js

```diff
@@ -81,6 +123,12 @@
             if (response.data.errors) {
                 return;
             }
+
+            // set default upgrade compute checkbox state one time only, so not to override user's choice
+            if (vm.nodesUpgrade.computeUpgradeEnabled === null) {
```

Comment:
This will still not work properly if you refresh the page after starting the upgrade. `computeUpgradeEnabled` will be reset to `null` and will switch to `true` on first model update. I think you should simply always use the `nodes_selected_for_upgrade` part of response (with no special null handling conditions).

Reply:
The purpose of having computeUpgradeEnabled set to null at first was because we only want to set the checkbox to true only one time at the beginning (before any node upgrade starts) in normal case (I'm not talking about the refresh case yet). The nodes_selected_for_upgrade value will not be available at that point. Also, we would be setting the computeUpgradeEnabled var every upgrade cycle, which will override the user's choice if the choice is different than the value in computeUpgradeEnabled.

Now I hadn't thought about the refresh case after starting the upgrade. I could use the nodes_selected_for_upgrade value in that case. I will look into this and make sure that both cases work.

Reply:
OK, I see that this is tricky case. The choice is persisted to the backend only after start of upgrade. The only way to remember user's choice across refreshes would be to use LocalStorage or something similar. I think this will be an overkill in this particular case.

Reply:
... but still we should prefer the value from status response if it is available to properly show the state of UI on refresh after upgrade was started.